Feature task6

AlexLukashevich95 opened the pull request:
-added new methods for adding products to shop and show them

AlexLukashevich95 committed:
-Feature-task6:
added new methods for adding products to shop.

AlexLukashevich95 committed:
Feature-task6:
-fixed add products to shop method
-added output products from shop functionality

AlexLukashevich95 committed:
Feature-task6:
-make code more clean

Evgnanii commented on src/com/lukashevich/shop/application/Menu.java:
```diff
@@ -87,16 +102,65 @@ private Address setAddress() {
-    private Long inputNumber(){
+
+    public Shop addProductsToShop(Shop shop) {
+        List<Product> productList = productController.getAllProducts();
```
добавь на бекенд метод который отдаст продукты в правильном виде по id магазина(или имени)

Evgnanii commented on src/com/lukashevich/shop/application/Menu.java:
```diff
@@ -87,16 +102,65 @@ private Address setAddress() {
+
+    public Shop addProductsToShop(Shop shop) {
+        List<Product> productList = productController.getAllProducts();
+        List<ProductShop> productShopList = shop.getProducts();
```
создай пустой список который будет хранить пары ключ значение. и добавляй в него id продуктов и количество. и вызывай метод бека который принимает в себя список этих пар и id магазина в который нужно добавить

AlexLukashevich95 replied:
Тут реализовано получение продуктов магазина в виде списка типа ProductShop. Сделано это для возможности расширения данных продуктов в магазине. К примеру можно добавить цену товара. В каждом магазине она может отличаться.

Evgnanii commented on src/com/lukashevich/shop/repository/impl/ShopRepositoryImpl.java:
```diff
@@ -21,4 +35,43 @@ public List<Shop> getAllShops() throws IOException {
+    }
+
+    @Override
+    public Shop addProductToShop(Shop shop) throws IOException {
```
такого метода не должно быть, должно быть достаточно метода saveShop. Заполнение объекта shop должно в сервисе происходить

Evgnanii commented on src/com/lukashevich/shop/repository/ShopRepository.java:
```diff
@@ -9,4 +10,10 @@ public interface ShopRepository {
+
+    Shop addProductToShop(Shop shop) throws IOException;
+
+    List<ProductShop> getProductsInShop(Shop shop);
```
тоже самое, не должно быть этого метода. Нужные данные из магазина ты достаешь в сервисе. Репозиторий магазина не должен возвращать продукты

Evgnanii commented on src/com/lukashevich/shop/repository/impl/BaseRepository.java:
```diff
@@ -42,13 +42,36 @@ public T save(T object) throws IOException {
+
+    public T getById(Long id) throws IOException {
+        List<T> objects = getAll();
+        T object = null;  //не особо понимаю , как правильно инициализировать. new Object и new T не подходят
```
а зачем тебе создавать этот объект? просто верни нужный объект, когда найдешь его, а не присваивай его значение новому объекту

AlexLukashevich95 replied:
не знаю как обработать возвращение значения вне условия с таким подходом. Просто возвращать null? Я помню ты говорил, что такой подход неправильный и надо избегать возвращение нула.

AlexLukashevich95 committed:
Feature-task6:
*Changed save method implementation

AlexLukashevich95 committed:
Feature-task6:
-changed Products in shop storing type;
-save objects with changed data but same id, but add a new ones

AlexLukashevich95 committed:
Merge remote-tracking branch 'origin/feature-task6' into feature-task6

# Conflicts:
#	src/com/lukashevich/shop/repository/impl/BaseRepository.java

AlexLukashevich95 committed:
Feature-task6:
-Added ProductShopDTO class
-added addProductToShop and getProductsInShop methods to shop service class

Evgnanii reviewed (approved):
LGTM!
